### PR TITLE
Provider util cleanup

### PR DESCRIPTION
**Pull request by RonnyPfannschmidt**
various cleanups that will help with the maintenance later on

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 1937e5fade8e0e51d7f3549e5f2625a0c5366693:

Everything seems all right :smile: .
*CFME QE Bot*

File: utils/providers.py [hunk 1]
```diff
@@ -93,10 +91,10 @@ def get_mgmt(provider_key, providers=None, credentials=None):
 
 
 def _get_provider_class_by_type(prov_type):
-    for _, class_dict in BaseProvider.type_mapping.iteritems():
-        for ptype, the_class in class_dict.iteritems():
-            if ptype == prov_type:
-                return the_class
+    for class_dict in BaseProvider.type_mapping.iteritems():
```
Comment (RonnyPfannschmidt) [inline utils/providers.py]: whops a bug  needs itervalues

**Comment by dajoRH** [comment]
Lint report for commit 19088966955ebbd45f54b148d5b24fb123b88bba:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by psav** [comment]
Squashy?

**Pull request merged**